cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -102,12 +102,40 @@ def check_for_upgrade_needed():
     add_rbac_roles()
     set_state('reconfigure.authentication.setup')
     remove_state('authentication.setup')
+    changed = snap_resources_changed()
+    if changed == 'yes':
+        set_upgrade_needed()
+    elif changed == 'unknown':
+        # We are here on an upgrade from non-rolling master
+        # Since this upgrade might also include resource updates eg
+        # juju upgrade-charm kubernetes-master --resource kube-any=my.snap
+        # we take no risk and forcibly upgrade the snaps.
+        # Forcibly means we do not prompt the user to call the upgrade action.
+        set_upgrade_needed(forced=True)
+
+
+
+def snap_resources_changed():
```

Comment:
flake8 has a whitespace concern here that will cause our builds to fail:
```
$ flake8 kubernetes_master.py
kubernetes_master.py:118:1: E303 too many blank lines (3)
```

Reply:
easy fix :) thanks